### DISCUSSION TITLE
The new version cannot open the database , SQLCipher aes-256-cbc

### DISCUSSION THREAD

**Discussion by monkeycc** [pawelsalawa/sqlitestudio#4532] [Q&A]
### Details

![微信截图_20221022162458](https://user-images.githubusercontent.com/6490927/197329677-6a8ab034-1b71-440b-98d4-9a6f8811ec5b.png)

The old version opens normally

![微信截图_20221022162918](https://user-images.githubusercontent.com/6490927/197329694-06fc84f0-67ff-4cf9-8b63-54d117968b97.png)

How to open a new version

**Comment by pawelsalawa** [comment]
Following the official SQLCipher documentation https://www.zetetic.net/sqlcipher/sqlcipher-api/ :
```
PRAGMA cipher = 'aes-256-cbc';
PRAGMA kdf_iter = '4000';
PRAGMA cipher_page_size = 1024;
PRAGMA cipher_use_hmac = OFF;
```

**Reply by monkeycc** [reply]
Thanks for your reply

But I'm sorry
It's still wrong
![nnnn](https://user-images.githubusercontent.com/6490927/197331699-cd4db517-c4e0-4fa2-89e7-5f8642a46b57.png)

```
PRAGMA cipher = 'aes-256-cbc';
PRAGMA kdf_iter = '4000';
PRAGMA cipher_page_size = 1024;
PRAGMA cipher_use_hmac = OFF;
```

**Reply by pawelsalawa** [reply]
Hm. Then I recommend to open it in older SQLiteStudio and decrypt it by opening SQL Editor window, set it's execution database to your SQLCipher database and execute:
```
ATTACH DATABASE 'decrypted.db' AS not_enrypted KEY '';
SELECT sqlcipher_export('not_enrypted');
DETACH DATABASE not_enrypted;
```
This way you have the `decrypted.db` database with no encyprion.
Then you can open it in new SQLiteStudio (using SQLCipher plugin) and set the password (encrypt) again:
```
ATTACH DATABASE 'encrypted.db' AS encrypted KEY 'your password';
SELECT sqlcipher_export('encrypted');
DETACH DATABASE encrypted;
```
Then you have the `encrypted.db` that uses recent SQLCipher, which should be easy to open in new SQLiteStudio.

**Reply by monkeycc** [reply]
Thank you for your help
